### ISSUE TITLE
How can I get headers from websocket client using ws, not socket.io?

### ISSUE THREAD

**Issue by tannineo** [nestjs/nest#813]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
In `web-sockets-controller.ts` we have
```ts
  public getConnectionHandler(
    context: WebSocketsController,
    instance: NestGateway,
    messageHandlers: MessageMappingProperties[],
    disconnect: Subject<any>,
    connection: Subject<any>,
  ) {
    const adapter = this.config.getIoAdapter();
    return client => {
      connection.next(client);
      context.subscribeMessages(messageHandlers, client, instance);

      const disconnectHook = adapter.bindClientDisconnect;
      disconnectHook &&
        disconnectHook.call(adapter, client, _ => disconnect.next(client));
    };
  }
```
and the func returned is used in `ws-adapter.ts`
```ts
  public bindClientConnect(server, callback: (...args) => void) {
    server.on(CONNECTION_EVENT, callback);
  }
```
The return func cannot accept a second arg `req` given by `ws`, which contains the http request of the connection, including ip and headers, which can be used for authentications and so on.

## Expected behavior
`bindClientConnect` should bind multiple args.

## Minimal reproduction of the problem with instructions
Use the `WSAdapter` given in `@nest/websockets` and try to get headers from the client.

## What is the motivation / use case for changing the behavior?
For some reason the client side of our team cannot use `socket.io` for the websocket communication, we use `ws` instead. When it comes to auth issues we find it difficult getting headers from the client in `handleConnection` of the gateway.

## Environment

<pre><code>
Nest version: 5.0.0

 
For Tooling issues:
- Node version: 8.11.3  
- Platform:  Mac 

Others:

</code></pre>

PS. I made a huge mistake by publishing this issue in [a wrong place](https://github.com/websockets/ws/issues/1407).

**Comment by tannineo** [comment]
@kamilmysliwiec Is there any possibility to change the returned callback func of `getConnectionHandler ` with minimum modification?

**Comment by tannineo** [comment]
#820 ready

**Comment by tannineo** [comment]
#821 
But I have some issue with CI, it cannot pass `npm ci`, but install and build work fine locally.

**Comment by kamilmysliwiec** [comment]
This feature just landed in the 5.3.0
